qrenderdoc/Windows/PipelineState/VulkanPipelineStateViewer.cpp

```diff
@@ -579,6 +579,33 @@ bool VulkanPipelineStateViewer::showNode(bool usedSlot, bool filledSlot)
   return false;
 }
 
+QString VulkanPipelineStateViewer::formatByteRange(const BufferDescription *buf,
+                                                   const VKPipe::BindingElement *descriptorBind)
+{
+  if(buf == NULL || descriptorBind == NULL)
+    return lit("-");
+  if(descriptorBind->byteSize == 0)
+  {
+    return QFormatStr("%1 - %2 %3")
+        .arg(descriptorBind->byteOffset)
+        .arg(descriptorBind->byteOffset)
+        .arg(tr("(empty view)"));
+  }
+  else if(descriptorBind->byteSize == UINT64_MAX)
+  {
+    return QFormatStr("%1 - %2 %3")
+        .arg(descriptorBind->byteOffset)
+        .arg(descriptorBind->byteOffset + (buf->length - descriptorBind->byteOffset))
+        .arg(tr("(whole size)"));
```

Comment:
You can use `tr` here directly if it's going to include some translatable text, instead of `QFormatStr` with a `tr` argument. `QFormatStr` is usually only when there's no text and all we're doing is formatting some numbers/parameters.

Also since this is vulkan-specific `(VK_WHOLE_SIZE)` is probably clearer than `(whole size)`, since it'd also make more sense when the offset is non-zero and the range is not actually the _whole_ size... but it is using `VK_WHOLE_SIZE` to indicate 'cover the rest of the buffer'. Similarly it would distinguish this case more clearly from e.g. covering bytes 0-256 of a 256 buffer explicitly without using the magic value, since that would fall into the last else here.

Reply:
OK, made it VK_WHOLE_SIZE.